pages/dashboard/locations/new.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { makeStyles } from '@material-ui/core';
+import { useRouter } from 'next/router';
+
+import { locationActions } from 'client/store/actions';
+import { AppStoreState } from 'client/store/reducers';
+import LocationForm from 'client/components/LocationForm';
+
+const useStyles = makeStyles(() => ({
+  responseDiv: {
+    margin: '15px 0',
+  },
+}));
+
+const NewLocation: React.FC = () => {
+  const router = useRouter();
+  const styles = useStyles();
+
+  const { error, loading } = useSelector((state: AppStoreState) => ({
+    error: state.locations.create.error,
+    loading: state.locations.create.loading,
+  }));
+  const dispatch = useDispatch();
+
+  const onSubmit = async data => {
+    await dispatch(locationActions.create(data));
+    router.replace('/dashboard/locations');
```

Comment:
same re replace vs push.  push is generally preferred.

Reply:
Yes, but here I'd argue that we should keep replace because when you create something you don't want to go back to the new page. Or do we maybe? (longer explanation in a comment before)